libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c

```diff
@@ -999,6 +992,9 @@ TEST( MQTT_Unit_API, ConnectMallocFail )
     int32_t i = 0;
     IotMqttError_t status = IOT_MQTT_STATUS_PENDING;
     IotMqttConnectInfo_t connectInfo = IOT_MQTT_CONNECT_INFO_INITIALIZER;
+    IotMqttSerializer_t serializer = IOT_MQTT_SERIALIZER_INITIALIZER;
+
+    _pAllocatedPingreq = NULL;
```

Comment:
Why does this need to be re-initialized?

Reply:
It doesn't